src/header/common/content_disposition.rs

```diff
@@ -21,27 +38,163 @@ pub enum DispositionType {
     /// Attachment implies that the recipient should prompt the user to save the response locally,
     /// rather than process it normally (as per its media type).
     Attachment,
-    /// Extension type.  Should be handled by recipients the same way as Attachment
-    Ext(String)
+    /// Used in *multipart/form-data* as defined in
+    /// [RFC7578](https://tools.ietf.org/html/rfc7578) to carry the field name and the file name.
+    FormData,
+    /// Extension type. Should be handled by recipients the same way as Attachment
+    Ext(String),
+}
+
+impl<'a> From<&'a str> for DispositionType {
+    fn from(origin: &'a str) -> DispositionType {
+        match origin.as_bytes().to_ascii_lowercase().as_slice() {
```

Comment:
I know it makes it more elegant, but I think it would be better to use non-allocating comparison using `to_ascii_lowercase`